### PR TITLE
Generate all Subsets in C++

### PR DESCRIPTION
**Pull request by deagleSC**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

C++ Program to generate and display all subsets of a given integer array using a Recursive backtracking approach in O(n) space complexity and O(2^n) time complexity, addressing the issue #1049

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Addressing #1049

### CODE REVIEW

File: C-Plus-Plus/other/subsets.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+        subsetArray[pos] = inputArray[pos]; 
+        GetSubsets(inputArray, subsetArray, pos+1); //Recursion tree branch 2
+    }
+}
+
+//DRIVER FUNCTION
+int main() {
+    int size;
+    cin >> size;
```
Comment (atarax665) [inline C-Plus-Plus/other/subsets.cpp]: Prompt user to enter the size of the array.

File: C-Plus-Plus/other/subsets.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        cin >> inputArray[i];
+    }
+
+    vector<int> subsetArray (size); //empty array for storing subsets
+    cout << "[";
+    GetSubsets(inputArray, subsetArray, 0); //prints the subsets
+    cout << "]";
+}
+
+//Sample IO
```
Comment (atarax665) [inline C-Plus-Plus/other/subsets.cpp]: Use multi-line comment. i.e ```/*............................*/```

**Review by atarax665 — CHANGES_REQUESTED** [review]
Format your code according to [C++ style guide](https://google.github.io/styleguide/cppguide.html). You can use [this](https://www.techiedelight.com/tools/clike).

File: C-Plus-Plus/other/subsets.cpp (hunk 2 above)
Comment (deagleSC) [reply C-Plus-Plus/other/subsets.cpp]: Made the changes, @atarax665

**Review by atarax665 — APPROVED** [review]

**Review by charlie219 — APPROVED** [review]

**Pull request merged**